projects/packages/password-checker/src/class-password-checker.php

```diff
@@ -0,0 +1,1303 @@
+
+		if ( ! $sections ) {
+			return $tests;
+		}
+
+		$sections = (array) $sections;
+		return array_intersect_key( $tests, array_flip( $sections ) );
+	}
+
+	/**
+	 * Provides the regular expression tester functionality.
+	 *
+	 * @param array $test_data the current test data.
+	 * @return Boolean does the test pass?
+	 */
+	protected function test_preg_match( $test_data ) {
+		$password = stripslashes( $this->password );
+
+		if ( isset( $test_data['trim'] ) ) {
+			$password = substr( $password, 1, -1 );
+		}
+
+		if ( ! preg_match( '/' . $test_data['pattern'] . '/u', $password ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Provides the comparison tester functionality.
+	 *
+	 * @param array $test_data the current test data.
+	 * @return Boolean does the test pass?
+	 */
+	protected function test_compare_to_list( $test_data ) {
+		$list_callback    = $test_data['list_callback'];
+		$compare_callback = $test_data['compare_callback'];
+
+		if (
+			! is_callable( array( $this, $list_callback ) )
+			|| ! is_callable( array( $this, $compare_callback ) )
+		) {
+			return false;
+		}
+
+		$list = call_user_func( array( $this, $list_callback ) );
+		if ( empty( $list ) ) {
+			return true;
+		}
+
+		return call_user_func( array( $this, $compare_callback ), $this->password, $list );
+	}
+
+	/**
+	 * Getter for the common password list.
+	 *
+	 * @return array common passwords.
+	 */
+	protected function get_common_passwords() {
+		return $this->common_passwords;
+	}
+
+	/**
+	 * Returns the widely known user data that can not be used in the password to avoid
+	 * predictable strings.
+	 *
+	 * @return array user data.
+	 */
+	protected function get_other_user_data() {
+		if ( empty( $this->user_id ) ) {
+			return array();
+		}
+
+		$user_data = get_userdata( $this->user_id );
+		if ( ! $user_data ) {
+			return array();
+		}
+
+		if ( isset( $user_data->ID ) ) {
+			$this->add_user_strings_to_test( get_user_meta( $user_data->ID, 'first_name', true ) );
+			$this->add_user_strings_to_test( get_user_meta( $user_data->ID, 'last_name', true ) );
+			$this->add_user_strings_to_test( get_user_meta( $user_data->ID, 'nickname', true ) );
+		}
+
+		if ( isset( $user_data->user_nicename ) ) {
+			$this->add_user_strings_to_test( $user_data->user_nicename );
+		}
+
+		if ( isset( $user_data->display_name ) ) {
+			$this->add_user_strings_to_test( $user_data->display_name );
+		}
+
+		if ( isset( $user_data->first_name ) ) {
+			$this->add_user_strings_to_test( $user_data->first_name );
+		}
+
+		if ( isset( $user_data->last_name ) ) {
+			$this->add_user_strings_to_test( $user_data->last_name );
+		}
+
+		if ( isset( $user_data->user_email ) ) {
+			$email_username = substr( $user_data->user_email, 0, strpos( $user_data->user_email, '@' ) );
+			$this->add_user_strings_to_test( $email_username, '.' );
+			$this->add_user_strings_to_test( $user_data->user_email );
+		}
+
+		return $this->user_strings_to_test;
```

Comment:
Probably doesn't matter much, but if this test is called multiple times this array will keep growing in size with duplicate values.

Reply:
That is an excellent point!
I've updated the code to better filter out emptyish strings and also passing the strings through array_unique().